prism-agent/service/wallet-api/src/main/scala/io/iohk/atala/agent/walletapi/util/KeyResolver.scala

```diff
@@ -1,30 +1,33 @@
 package io.iohk.atala.agent.walletapi.util
 
-import io.iohk.atala.agent.walletapi.crypto.Apollo
-import io.iohk.atala.agent.walletapi.crypto.ECKeyPair
+import io.iohk.atala.shared.crypto.Apollo
+import io.iohk.atala.shared.crypto.Secp256k1KeyPair
 import io.iohk.atala.agent.walletapi.model.KeyManagementMode
 import io.iohk.atala.agent.walletapi.model.ManagedDIDState
 import io.iohk.atala.agent.walletapi.model.WalletSeed
 import io.iohk.atala.agent.walletapi.storage.DIDNonSecretStorage
 import io.iohk.atala.agent.walletapi.storage.WalletSecretStorage
-import io.iohk.atala.castor.core.model.did.EllipticCurve
 import io.iohk.atala.castor.core.model.did.PrismDID
 import io.iohk.atala.shared.models.WalletAccessContext
 import zio.*
 
 class KeyResolver(apollo: Apollo, nonSecretStorage: DIDNonSecretStorage, walletSecretStorage: WalletSecretStorage) {
-  def getKey(state: ManagedDIDState, keyId: String): RIO[WalletAccessContext, Option[ECKeyPair]] = {
+  def getKey(state: ManagedDIDState, keyId: String): RIO[WalletAccessContext, Option[Secp256k1KeyPair]] = {
```

Comment:
Is this just the Master key?

Reply:
It should be able to get any key. Just overload it for convenience.

Reply:
could it be refactored to the typed method, so the type of the key pair will be set explicitly?
getKey[Secp256k1KeyPair](...)